ethernet driver net_pkt api switch and bug fix

tbursztyka opened the pull request.

Commit:
drivers/ethernet: Switch stellaris driver to new net_pkt API

Allocating all the necessary buffer at once.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ethernet: Removing unused variables in stellaris driver

In was generating these warnings:
eth_stellaris.c:66:8: warning: unused variable ‘eth_hdr’
eth_stellaris.c:65:8: warning: unused variable ‘head_len_left’

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ethernet: Fix net API usage in smsc911x driver

Fixes #13484

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

jukkar commented on drivers/ethernet/eth_stellaris.c:
```diff
@@ -175,38 +183,34 @@ static int eth_stellaris_rx_pkt(struct device *dev, struct net_pkt *pkt)
+	pkt = eth_stellaris_rx_pkt(dev, iface);
 	if (!pkt) {
-		LOG_ERR("Could not allocate pkt");
+		LOG_ERR("Failed to read data");
```
Minor thing, but I think we should say "Failed to write data" here.

tbursztyka replied:
at this point we are actually trying to read data from the hw, sure we write it back to pkt. But it's same same I would say.

jukkar commented on drivers/ethernet/eth_smsc911x.c:
```diff
@@ -578,12 +578,13 @@ static struct net_pkt *smsc_recv_pkt(struct dehive *dev, u32_t pkt_size)
 	/* Discard FCS */
 	{
-		u32_t dummy = SMSC9220->RX_DATA_PORT;
+		u32_t __unused dummy = SMSC9220->RX_DATA_PORT;
```
Can't we just remove the dummy block here?

tbursztyka replied:
No, this line does flush the rx data port register by reading it. Original code needed it, so I guess this is still valid.

jukkar replied:
Ah, ok then.

Commit:
drivers/net: Switch slip driver to new net_pkt api

Replace "frag" keyword to "buf", as frag is reserved for packet
fragmentation now.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

jukkar reviewed (approved):
LGTM
